definitions/npm/cookie_v0.3.x/flow_v0.58.x-/cookie_v0.3.x.js

```diff
@@ -0,0 +1,22 @@
+type SerializeOptions = {
+  domain?: ?string,
+  encode?: ?((stringToDecode: string) => string),
+  expires?: ?Date,
+  httpOnly?: ?boolean,
+  maxAge?: ?number,
+  path?: ?string,
+  sameSite?: ?(boolean | 'lax' | 'strict'),
+  secure?: ?boolean,
+};
+
+type ParseOptions = {
+  decode?: ?((stringToDecode: string) => *),
+};
+
+declare module 'cookie' {
+  declare module.exports: {
+    serialize(name: string, val: string, options: ?SerializeOptions): string,
+    // eslint-disable-next-line flowtype/no-weak-types
+    parse(data: string, options: ?ParseOptions): Object,
```

Comment:
`Object` is pretty useless. I'd go with someone like `{ [key: string]: string }` if possible.

Reply:
Type of value of object depend of decode options. I think it should be something like this:

```
type ParseOptions<ParserOutput> = {
  decode?: ?((stringToDecode: string) => ParserOutput),
};

declare module 'cookie' {
...
    parse<ParserOutput>(data: string, options: ?ParseOptions<ParserOutput>): { [key: string]: ParserOutput },
...
};
```

I'm not sure what is the best solution. If decode options is not defined, output is always string. But user have option to create own decoder with any response type he choose.

I think there are two options.
1. Set return type of decode option to string, which is logical because you are working with cookies. I will make definition clearer and simplier.
2. Use generic type as wrote in this comment.

Reply:
Unless there's a strong use case for it, I recommend keeping things simple. You could reference this comment in the libdef, in case it becomes necessary, the work doesn't need to be redone.

Reply:
Ok I updated my PR